src/main/java/com/example/onlinebookstore/dto/UserLoginRequestDto.java

```diff
@@ -0,0 +1,17 @@
+package com.example.onlinebookstore.dto;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Size;
+
+public record UserLoginRequestDto(
+        @NotEmpty
+        @Size(min = 8, max = 20)
```

Comment:
20 chars for email seems too restrictive. I'm sure there are a lot of longer emails.